core/cas-server-core-services/src/main/java/org/apereo/cas/services/RegexRegisteredService.java

```diff
@@ -42,18 +40,13 @@ public boolean matches(final Service service) {
     @Override
     public boolean matches(final String serviceId) {
         if (this.servicePattern == null) {
```

Comment:
This wont work. You will need to make sure the given pattern is valid; it's not appropriate to replace it with some other pattern in case it's invalid.